### ISSUE TITLE
Unable to update tags for existing SQS queue

### ISSUE THREAD

**Issue by rj425** [aws-controllers-k8s/community#1541]
**Describe the bug**
I have a SQS queue that has few tags already. On adding new tags to the `.spec.tags`, i can see the new tags appearing in manifest and everything. But, the AWS console shows otherwise. New tags are not reflected.

```
tags:
  namespace: {{ $.Values.namespace }}
  app: {{ $.Values.app_name }}   
  abc:v1:costAllocation1: {{ $.Values.team_id }}
  abc:v1:costAllocation2: {{ $.Values.namespace }}
  abc:v1:costAllocation3: {{ $.Values.app_name }} 
```

**Steps to reproduce**
- Create a queue
- Add new tags to that queue
- New tags don't appear in AWS console.

**Expected outcome**
New tags should appear in AWS console.

**Comment by rj425** [comment]
`/remove-lifecycle stale`

**Comment by rj425** [comment]
Also, recently noticed a similar behavior with resource access policies.
For an existing SQS Queue, we updated the access policy within that resource. And the SQS controller fails to update the queue policy noticed in AWS console.

@A-Hilaly Can this please be looked at?
Thanks.

**Comment by jaypipes** [comment]
@rj425 so, the Policy attribute should definitely be being updated. Can you please paste the YAML (redacted for any account-specific stuff) that you are using before and after Spec update so I can add a reproducible test case?

As for tag support, the SQS controller does not yet support tags (either setting or updating). I will work on this feature shortly.

**Comment by rj425** [comment]
Hi @jaypipes 

Thank you for taking this up.

**Resource Definition before update**
```
apiVersion: sqs.services.k8s.aws/v1alpha1
kind: Queue
metadata:
  annotations:
    argocd.argoproj.io/tracking-id: 'dev-simple-app:sqs.services.k8s.aws/Queue:rjain/vega-simple-app-queue1'
  name: vega-simple-app-queue1
  namespace: rjain
spec:
  delaySeconds: '0'
  policy: |
    {

      "Version": "2008-10-17",
      "Id": "__default_policy_ID",
      "Statement": [
        {
          "Sid": "__owner_statement",
          "Effect": "Allow",
          "Principal": {
            "AWS": "1111111111111111"
          },
          "Action": [
            "SQS:*"
          ],
          "Resource": "arn:aws:sqs:*:1111111111111111:vega-simple-app-queue1"
        },
        {
          "Sid": "__sender_statement",
          "Effect": "Allow",
          "Principal": {
            "AWS": [
              "arn:aws:iam::1111111111111111:role/vega-simple-app"
            ]
          },
          "Action": [
            "SQS:SendMessage"
          ],
          "Resource": "arn:aws:sqs:*:1111111111111111:vega-simple-app-queue1"
        },
        {
          "Sid": "__receiver_statement",
          "Effect": "Allow",
          "Principal": {
            "AWS": [
              "arn:aws:iam::1111111111111111:role/vega-simple-app"
            ]
          },
          "Action": [
            "SQS:ChangeMessageVisibility",
            "SQS:DeleteMessage",
            "SQS:ReceiveMessage"
          ],
          "Resource": "arn:aws:sqs:*:1111111111111111:vega-simple-app-queue1"
        }
      ]
    }
  queueName: vega-simple-app-queue1
  tags:
    app: simple-app
    namespace: rjain
    'xyz:v1:costAllocation1': vega-core-infra
    'xyz:v1:costAllocation2': rjain
    'xyz:v1:costAllocation3': simple-app
```

**Resource definition after update**
```
apiVersion: sqs.services.k8s.aws/v1alpha1
kind: Queue
metadata:
  annotations:
    argocd.argoproj.io/tracking-id: 'dev-simple-app:sqs.services.k8s.aws/Queue:rjain/vega-simple-app-queue1'
  name: vega-simple-app-queue1
  namespace: rjain
spec:
  delaySeconds: '0'
  policy: |
    {

      "Version": "2008-10-17",
      "Id": "__default_policy_ID",
      "Statement": [
        {
          "Sid": "__owner_statement",
          "Effect": "Allow",
          "Principal": {
            "AWS": "1111111111111111"
          },
          "Action": [
            "SQS:*"
          ],
          "Resource": "arn:aws:sqs:*:1111111111111111:vega-simple-app-queue1"
        },
        {
          "Sid": "__sender_statement",
          "Effect": "Allow",
          "Principal": {
            "AWS": [
              "arn:aws:iam::1111111111111111:role/vega-simple-app"
            ]
          },
          "Action": [
            "SQS:SendMessage"
          ],
          "Resource": "arn:aws:sqs:*:1111111111111111:vega-simple-app-queue1"
        },
        {
          "Sid": "__receiver_statement",
          "Effect": "Allow",
          "Principal": {
            "AWS": [
              "arn:aws:iam::1111111111111111:role/vega-simple-app"
            ]
          },
          "Action": [
            "SQS:ChangeMessageVisibility"
          ],
          "Resource": "arn:aws:sqs:*:1111111111111111:vega-simple-app-queue1"
        }
      ]
    }
  queueName: vega-simple-app-queue1
  tags:
    app: simple-app
    namespace: rjain
    'xyz:v1:costAllocation1': vega-core-infra
    'xyz:v1:costAllocation2': rjain
    'xyz:v1:costAllocation3': simple-app
```

**Comment by jaypipes** [comment]
@rj425 OK! This is now fixed in the [v0.0.07](https://github.com/aws-controllers-k8s/sqs-controller/releases/tag/v0.0.7) release of the SQS controller :) Please close this issue out after verifying everything is properly working for you.

**Comment by rj425** [comment]
@jaypipes Thank you so much for taking this up.
Does this fix also address the issue of resource policy not updating the actual AWS resource ([issue](https://github.com/aws-controllers-k8s/community/issues/1541#issuecomment-1481242900)).

I will test out this release.
Thanks again.

**Comment by jaypipes** [comment]
@rj425 sorry for delayed response. I have just [added an e2e test](https://github.com/aws-controllers-k8s/sqs-controller/pull/27) that updates queue attributes and verifies the changes are properly reflected in the queue on the AWS API side. So, yes, this should indeed be closed out with the latest (v1.0.0) sqs-controller release :)

**Comment by jaypipes** [comment]
Completed.
